Content.Server/Weapons/Reflect/ReflectSystem.cs

```diff
@@ -23,4 +26,16 @@ private void DisableReflect(EntityUid uid, ReflectComponent comp, ref EnergySwor
         comp.Enabled = false;
         Dirty(comp);
     }
-}
\ No newline at end of file
+
+    private void ShieldEnableReflect(EntityUid uid, ReflectComponent comp, ref EnergyShieldActivatedEvent args)
+    {
+        comp.Enabled = true;
+        Dirty(comp);
+    }
+
+    private void ShieldDisableReflect(EntityUid uid, ReflectComponent comp, ref EnergyShieldDeactivatedEvent args)
+    {
+        comp.Enabled = false;
+        Dirty(comp);
+    }
```

Comment:
```suggestion
```

Same with this, as long as reflectcomp is on the yml you shouldn't need this.

Reply:
Without this, Reflect does not recognize when it is on and when it is not. Without it, you can reflect rays even if the shield is off.

Reply:
This is fine then